docs/debug.md

```diff
@@ -0,0 +1,112 @@
+This is required, otherwise the *backend* and *cluster-agent* operators won't be able to fetch the secret from a different _namespace_.
+
+To install it into the Kubernetes cluster:
+
+```shell
+ make deploy-postgresql
+```
+
+### Access Postgres outside the cluster
+
+To be able to access the database from outside the cluster you need to forward it locally:
+
+```shell
+$ kubectl port-forward --namespace gitops-postgresql svc/gitops-postgresql-staging 5432:5432 &
+# Press Enter
+# Don't forget to kill it when you are finished: 'fg' and then 'CTRL+C'
+```
+
+Then run a `psql` command, to verify your access:
+
+```shell
+psql postgresql://postgres:$PASSWORD@127.0.0.1:5432/postgres -c "select now()"
+```
+
+_Note: If you are using a plain-text hardcoded password, replace `$PASSWORD` var with yours_.
+
+Expected Output:
+
+```shell
+Handling connection for 5432
+              now
+-------------------------------
+ 2022-01-25 15:37:07.826248+00
+(1 row)
+```
+
+### Access Postgres within the cluster
+
+From within the cluster, the `coredns` pod will be able to resolve the _service_ name, and you will also need to fetch the `secret`.
+For example, the `gitops-service-backend` has the following configuration snippet into its deployment YAML:
+
+```yaml=
+        env:
+          - name: DB_ADDR
+            value: gitops-postgresql-staging.gitops-postgresql
+          - name: DB_PASS
+            valueFrom:
+              secretKeyRef:
+                name: gitops-postgresql-staging
+                key: postgresql-password
+        image: quay.io/pgeorgia/gitops-service:latest
+```
+
+When the `manager` container is running, it fetches those ENV variables and uses them to connect to the database.
+For more information, read the [connectToDatabase] function.
+
+#### Verify the DNS is working
+
+The Pods within the cluster should be able to resolve the name of the services.
+
+```shell
+kubectl apply -f https://k8s.io/examples/admin/dns/dnsutils.yaml
+```
+
+To make sure the coredns works correctly within the cluster, try this:
+
+```shell
+$ kubectl exec -i -t dnsutils -- nslookup gitops-postgresql-staging-headless.gitops-postgresql
+
+Server:		10.96.0.10
+Address:	10.96.0.10#53
+
+Name:	gitops-postgresql-staging-headless.gitops-postgresql.svc.cluster.local
+Address: 10.1.0.6
+```
+
+```shell
+$ kubectl exec -i -t dnsutils -- nslookup gitops-postgresql-staging.gitops-postgresql
+
+Server:  10.96.0.10
+Address: 10.96.0.10#53
+
+Name: gitops-postgresql-staging.gitops-postgresql.svc.cluster.local
+Address: 10.96.236.173
+```
+
+#### Connect to the database
+
+Spawn a Postgres Pod and try to access the database using the service.
+
+```shell
+kubectl run pgsql-postgresql-client --rm --tty -i --restart='Never' --namespace gitops-postgresql --image docker.io/bitnami/postgresql:11.7.0-debian-10-r9 --env="PGPASSWORD=$PASSWORD" --command -- psql testdb --host gitops-postgresql-staging-headless.gitops-postgresql -U postgres -d postgres -p 5432
```

Comment:
This isn't a supported image on OpenShift.
The equivalent of `registry.redhat.io/rhel8/postgresql-10:latest` should get you going. You'll probably need this on staging too initially till we figure out how to get an AWS RDS.

Reply:
@sbose78 do you happen to know what the relationship is between that and https://catalog.redhat.com/software/containers/rhel8/postgresql-12/5db133bd5a13461646df330b?container-tabs=overview, which Panos posted earlier? Are they both supported? It seems like the catalog.redhat.com contains a much newer version, but I'm not especially familiar with how Red Hat packages image and what the varying degrees of support are..

Reply:
https://github.com/openshift/library/blob/master/official/postgresql/imagestreams/postgresql-rhel.json#L47

Yeah, both are supported. This is a good way to keep a eye on what goes into every OCP release https://github.com/openshift/library/tree/master/official 

These would manifest as reference-able imagestreams in the `openshift` namespace but consumable by all users in all namespaces.